File: Tracks/typescript/bob/bob.ts
class Bob {
    private isYelling = (message: string): boolean =>
        message.match(/[A-Z]/g) !== null && message === message.toUpperCase();

    hey(message: string) {
        const parsedMessage = message.trim();
        if (parsedMessage.length === 0) return "Fine. Be that way!";
        const isYelling = this.isYelling(parsedMessage);
        if (parsedMessage.endsWith("?")) {
            return isYelling ? "Calm down, I know what I'm doing!" : "Sure.";
        }
        return isYelling ? "Whoa, chill out!" : "Whatever.";
    }
}

export default Bob;

File: Tracks/csharp/bob/Bob.cs
using System.Linq;
using System.Text.RegularExpressions;

public static class Bob
{
    private static bool IsYelling(this string statement) => new Regex("[A-Z]").IsMatch(statement) && statement == statement.ToUpper();

    private static bool IsSilent(this string statement) => string.IsNullOrWhiteSpace(statement);

    private static bool IsQuestion(this string statement) => statement.TrimEnd().Last() == '?';

    public static string Response(string statement)
    {
        if(statement.IsSilent())
        {
            return "Fine. Be that way!";
        }

        if(statement.IsQuestion())
        {
            return statement.IsYelling() ? "Calm down, I know what I'm doing!" : "Sure.";
        }

        return statement.IsYelling() ? "Whoa, chill out!" : "Whatever.";
    }
}